grpc/server/test/action_service_impl_test.cpp

```diff
@@ -0,0 +1,274 @@
+#include <gmock/gmock.h>
+#include <string>
+
+// TODO remove this include
+#include "action/action.h"
+#include "action/action_service_impl.h"
+#include "action/mocks/action_mock.h"
+
+using testing::NiceMock;
+using testing::Return;
+
+using MockAction = NiceMock<dronecore::testing::MockAction>;
+using ActionServiceImpl = dronecore::backend::ActionServiceImpl<MockAction>;
+
+using ActionResult = dronecore::rpc::action::ActionResult;
+
+std::string armAndGetTranslatedResult(dronecore::Action::Result arm_result);
+std::string takeoffAndGetTranslatedResult(dronecore::Action::Result takeoff_result);
+std::string landAndGetTranslatedResult(dronecore::Action::Result land_result);
+
+TEST(ActionServiceImpl, armSuccessIsCorrectlyTranslated)
```

Comment:
Many tests here, but if you look closely, you'll see that it checks all the cases for those commands. This ensures that we don't modify `ActionResult` in `action/action.h` without making sure the conversion to the gRPC enum is correct, so it will probably avoid weird bugs.

Reply:
Oh wow, tedious! But I appreciate the tests but I doubt that many tests like this will scale.

Doesn't the comiler complain if you don't cover all enum fields in a switch case? So if you don't add the `default` case, it would tell you when a new enum was added that is not checked for?

Reply:
The real problem is that we have to translate `Action::Result::SUCCESS` into the gRPC enum as defined by the proto file. And in this case we do `static_cast`, and we are happy because luckily, `UNKNOWN` is "0" on both sides. 

Now say you swap two of them in `action/action.h`, I'm quite sure that might pass a review and get into the code. And then, some customers might receive the messages inverted. And since some messages are pretty rare, good luck for debugging :-). I'd rather have a few more unit tests like this.

Maybe there is a less verbose way to write them in C++, or maybe that might be a situation for using macros, I don't know. But I believe those tests are valuable, and still much easier to write as unit tests than doing integration testing.

Reply:
Oh, and note that those unit tests are really high level, so I would not expect them to change, even if we change the implementation of the `ActionServiceImpl` functions (that's actually a goal of unit tests, I think: they only depend on the public interface).

So it takes some time to write them, but I don't expect to have to maintain them, like, at all :D.

Reply:
In the Yuneec SDK Android wrapper I used C macros:
```
#define CASE(_name) \
case YuneecSt16::Result::_name: \
    enumStateField = menv->GetStaticFieldID( \
        ResultEnum, #_name, "Lcom/yuneec/sdk/YuneecSt16$Result$ResultID;"); \
break;

        switch (result) {
                CASE(SUCCESS)
                CASE(BUSY)
                CASE(ALREADY_PAIRED)
                CASE(NOT_PAIRED)
                CASE(UNSUPPORTED_PLATFORM)
                CASE(UNKNOWN)
        }
#undef CASE
```

Reply:
Yeah, that may be a solution, but I must say I am not comfortable reading macros. Another way would be to create a C++ map so that we can `for`-loop on the different values we have, and also use some lambdas to choose the function (now it is repeated for `arm()`, `land()`, `takeoff()`).

But all of them make the code more complex, and if a test fails, you have to go read those constructions to understand what happend. Whereas the hardcoded ones (if they scale, somehow) are very straightforward to understand when they break.

I would suggest we keep the hardcoded ones and wait to see how it evolves, if that's fine for you.

Reply:
Fair point, ok sure, let's do that for now.